subxt/src/config/mod.rs

```diff
@@ -129,51 +129,24 @@ pub trait Header: Sized + Encode + Decode {
 #[cfg(feature = "substrate-compat")]
 mod substrate_impls {
     use super::*;
-    use primitive_types::{H256, U256};
 
-    impl<N, H> Header for sp_runtime::generic::Header<N, H>
+    impl<T: sp_runtime::traits::Header> Header for T
     where
-        Self: Encode + Decode,
-        N: Copy + Into<U256> + Into<u64> + TryFrom<U256>,
-        H: sp_runtime::traits::Hash + Hasher,
+        u64: From<<T as sp_runtime::traits::Header>::Number>,
```

Comment:
```suggestion
        <T as sp_runtime::traits::Header>::Number: Into<u64>,
```

Little more consistent with the trait definition and clearer imo :)